pkg/vpcmodel/jsonOutput.go

```diff
@@ -108,3 +124,75 @@ func getConnLinesForSubnetsConnectivity(conn *VPCsubnetConnectivity) []connLine
 	sortConnLines(connLines)
 	return connLines
 }
+
+type allSemanticDiff struct {
+	SemanticDiff []diffLine `json:"semantic_diff"`
+}
+
+func getDiffLines(configsDiff *diffBetweenCfgs) []diffLine {
+	diffLines := getDirectionalDiffLines(configsDiff.cfg1ConnRemovedFrom2, true)
+	diffLines = append(diffLines, getDirectionalDiffLines(configsDiff.cfg2ConnRemovedFrom1, false)...)
+	return diffLines
+}
+
+func getDirectionalDiffLines(connectDiff connectivityDiff, thisMinusTheOther bool) []diffLine {
+	diffLines := []diffLine{}
+	for src, endpointConnDiff := range connectDiff {
+		for dst, connDiff := range endpointConnDiff {
+			var diffStr string
+			if thisMinusTheOther {
+				diffStr = getDiffStrThis(connDiff.diff)
+			} else {
+				diffStr = getDiffStrOther(connDiff.diff)
+			}
+			diffLines = append(diffLines, diffLine{diffStr, src, dst, common.ConnToJSONRep(connDiff.conn1), common.ConnToJSONRep(connDiff.conn2)})
+		}
+	}
+
+	sortDiffLines(diffLines)
+	return diffLines
+}
+
+func sortDiffLines(diffLines []diffLine) {
+	sort.Slice(diffLines, func(i, j int) bool {
+		if diffLines[i].Diff != diffLines[j].Diff {
+			return diffLines[i].Diff < diffLines[j].Diff
+		}
+		if diffLines[i].Src.Name() != diffLines[j].Src.Name() {
+			return diffLines[i].Src.Name() < diffLines[j].Src.Name()
+		}
+		return diffLines[i].Dst.Name() < diffLines[j].Dst.Name()
+	})
+}
+
+func getDiffStrThis(diff DiffType) string {
+	switch diff {
+	case missingSrcEP:
+		return "source removed"
+	case missingDstEP:
+		return "destination removed"
+	case missingSrcDstEP:
+		return "source and destination removed"
+	case missingConnection:
+		return "connection removed"
+	case changedConnection:
+		return connectionChanged
+	}
+	return ""
+}
+
+func getDiffStrOther(diff DiffType) string {
+	switch diff {
+	case missingSrcEP:
+		return "source added"
+	case missingDstEP:
+		return "destination added"
+	case missingSrcDstEP:
+		return "source and destination added"
+	case missingConnection:
+		return "connection added"
+	case changedConnection:
+		return connectionChanged
```

Comment:
define hard-coded strings as `const` values.. cannot be reused with const strings from other formats?

Reply:
done